appservice/src/intTest/groovy/nu/yona/server/DisclosureTest.groovy

```diff
@@ -1,182 +1,144 @@
+			messagesRichard[0].url != null
+			messagesRichard[0]._links.requestDisclosure == null
+
+			responseBob.status == 200
+			responseBob.responseData?._embedded?.goalConflictMessages
+			def messagesBob = responseBob.responseData._embedded.goalConflictMessages
+			messagesBob.size() == 1
+			messagesBob[0].nickname == richard.nickname
+			messagesBob[0].goalName == "news"
+			messagesBob[0].url == null
+			messagesBob[0]._links.requestDisclosure
+
+		cleanup:
+			appService.deleteUser(richard)
+			appService.deleteUser(bob)
 	}
 
-	def 'Richard checks his anonymous messages and finds the disclosure request'(){
+	def 'Richard receives Bob\'s request for disclosure of a message'()
+	{
 		given:
+			def richardAndBob = addRichardAndBobAsBuddies()
+			def richard = richardAndBob.richard
+			def bob = richardAndBob.bob
+			analysisService.postToAnalysisEngine(richard, ["Gambling"], "http://www.poker.com")
+			def discloseRequestURL = appService.getAnonymousMessages(bob).responseData._embedded.goalConflictMessages[0]._links.requestDisclosure.href
 
 		when:
-		def response = appService.getAnonymousMessages(richardQuin.url, richardQuin.password)
-		if(response.status == 200 && response.responseData._embedded.discloseRequestMessages) {
-			goalDiscloseRequestMessage1AcceptURL = response.responseData._embedded.discloseRequestMessages[0]._links.accept.href
-			goalDiscloseRequestMessage2RejectURL = response.responseData._embedded.discloseRequestMessages[1]._links.reject.href
-		}
+			def response = appService.postMessageActionWithPassword(discloseRequestURL, [ : ], bob.password)
 
 		then:
-		response.status == 200
-		response.responseData._embedded.discloseRequestMessages
-		response.responseData._embedded.discloseRequestMessages.size() == 2
-		response.responseData._embedded.discloseRequestMessages[1].nickname == bobDunn.nickname
-		response.responseData._embedded.discloseRequestMessages[1].targetGoalConflictMessage
-		response.responseData._embedded.discloseRequestMessages[1].targetGoalConflictMessage.goalName == "gambling"
-		//TODO response.responseData._embedded.discloseRequestMessages[1].targetGoalConflictMessage.creationTime
-		response.responseData._embedded.discloseRequestMessages[1]._links.related.href == richardQuinGoalConflictMessage1URL
-		goalDiscloseRequestMessage1AcceptURL
-		goalDiscloseRequestMessage2RejectURL
+			response.status == 200
+			def getRichardAnonMessagesResponse = appService.getAnonymousMessages(richard)
+			getRichardAnonMessagesResponse.status == 200
+			getRichardAnonMessagesResponse.responseData?._embedded?.discloseRequestMessages
+			def discloseRequestMessages = getRichardAnonMessagesResponse.responseData._embedded.discloseRequestMessages
+			discloseRequestMessages.size() == 1
+			discloseRequestMessages[0].targetGoalConflictMessage.goalName == "gambling"
+			discloseRequestMessages[0].targetGoalConflictMessage.creationTime > (System.currentTimeMillis() - 50000) // TODO Use standard date/time format
```

Comment:
Where do you take that from? Based on http://stackoverflow.com/a/15952652, I would think we can use a more readable format.